src/generic_rosenbrock.jl

```diff
@@ -0,0 +1,941 @@
+       9.7998186780974000 11.935792886603180 3.6738749290132010 14.807828541095500 0.8318583998690680]
+    b=[6.4562170746532350,-4.853141317768053,9.7653183340692600,2.0810841772787230,0.6603936866352417,0.6000000000000000]
+    gamma=0.2500000000000000
+    d=[0.2500000000000000,0.0836691184292894,0.0544718623516351,-0.3402289722355864,0.0337651588339529,-0.0903074267618540]
+    c=[0                 ,0.1453095851778752,0.3817422770256738,0.6367813704374599,0.7560744496323561,0.9271047239875670]
+    RosenbrockFixedTableau(a,C,b,gamma,d,c)
+end
+
+"""
+    @RosenbrockW6S4OS(part)
+
+Generate code for the RosenbrockW6S4OS method.
+`part` should be one of `:tableau`, `:cache`, `:init`, `:performstep`.
+`@RosenbrockW6S4OS(:tableau)` should be placed in `tableaus/rosenbrock_tableaus.jl`.
+`@RosenbrockW6S4OS(:cache)` should be placed in `caches/rosenbrock_caches.jl`.
+`@RosenbrockW6S4OS(:init)` and `@RosenbrockW6S4OS(:performstep)` should be
+placed in `perform_step/rosenbrock_perform_step.jl`.
+"""
+macro RosenbrockW6S4OS(part)
+    tab=RosenbrockW6S4OSTableau()
+    tabmask=_masktab(tab)
+    algname=:RosenbrockW6S4OS
+    tabname=:RosenbrockW6S4OSConstantCache
+    tabstructname=:RosenbrockW6S4OSConstantCache
+    cachename=:RosenbrockW6SCache
+    constcachename=:RosenbrockW6SConstantCache
+    n_normalstep=length(tab.b)-1
+    if part.value==:tableau
+        #println("Generating const cache")
+        tabstructexpr=gen_tableau_struct(tabmask,tabstructname)
+        tabexpr=gen_tableau(tab,tabstructexpr,tabname)
+        return esc(quote $([tabstructexpr,tabexpr]...) end)
+    elseif part.value==:cache
+        #println("Generating cache")
+        constcacheexpr,cacheexpr=gen_cache_struct(tabmask,cachename,constcachename)
+        algcacheexpr=gen_algcache(cacheexpr,cachename,constcachename,algname,tabname)
+        return esc(quote $([constcacheexpr,cacheexpr,algcacheexpr]...) end)
+    elseif part.value==:init
+        #println("Generating initialize")
+        return esc(gen_initialize(cachename,constcachename))
+    elseif part.value==:performstep
+        #println("Generating perform_step")
+        constperformstepexpr=gen_constant_perform_step(tabmask,constcachename,n_normalstep)
+        performstepexpr=gen_perform_step(tabmask,cachename,n_normalstep)
+        return esc(quote $([constperformstepexpr,performstepexpr]...) end)
+    else
+        println("Unknown parameter!")
+    end
+end
+
+"""
+    Ros4dummyTableau()
+
+Generate a dummy tableau for ROS4 methods. It can be considered as performing elementwise OR to the masks 
+of those sepecific tableaus: `Ros4dummyTableau()==_masktab(RosShamp4Tableau()) OR _masktab(Veldd4Tableau()) OR ...`
+ROS4 methods have the property of a4j==a3j so a is a 3*3 matrix instead of a 4*4 matrix and c is a 1*3 vector instead of a 1*4 vector.
+"""
+function Ros4dummyTableau()#create a tabmask for all ROS4 methods where false->0,true->non-0
```

Comment:
I wonder if the masking can be done at runtime. For instance, add `map(!iszero, xxx)` to the `RosenbrockAdaptiveTableau` constructor and fields to hold that information. Of course, this is not a priority.

Reply:
Yes, that can be done by `_masktab`. The point is, I prefer to specify a generic tab for a group of methods for places where generic characteristics are needed (like perform_step, tableau struct and cache struct) while not emphasizing any specific tableau.